Library/OcCpuLib/FrequencyDetect.c

```diff
@@ -225,6 +228,57 @@ InternalCalculateTSCFromPMTimer (
   return TSCFrequency;
 }
 
+UINT64
+InternalCalculateTSCFromApplePlatformInfo (
+  OUT  UINT64  *FSBFrequency  OPTIONAL
+  )
+{
+  EFI_STATUS                             Status;
+  APPLE_PLATFORM_INFO_DATABASE_PROTOCOL  *PlatformInfo;
+  UINT32                                 Size;
+  UINT64                                 FsbFreq;
+
+  Status = gBS->LocateProtocol (
+    &gApplePlatformInfoDatabaseProtocolGuid,
+    NULL,
+    (VOID **) &PlatformInfo
+    );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCCPU: Failed to locate ApplePlatformInfo protocol - %r\n", Status));
+    return 0;
+  }
+
+  Status = PlatformInfo->GetFirstDataSize (
+    PlatformInfo,
+    &gAppleFsbFrequencyPlatformInfoGuid,
+    &Size
+    );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCCPU: Failed to get first data size - %r\n", Status));
```

Comment:
Actually we should use a fallback mechanism when reading via automatic FirstData discovery fails. It is [here](https://github.com/acidanthera/mac-efi-firmware/blob/759669b625c7d4b6d1d09afd06dd20a0bce8b353/EfiPkg/Universal/AppleSmbiosDxe/AppleSmbios.c#L614) in the ref.

I.e. get the HOB and then use it manually. GetFirstGuidHob and GET_GUID_HOB_DATA are implemented by [HobLib](https://github.com/acidanthera/audk/blob/master/MdePkg/Include/Library/HobLib.h).